java/ql/src/Security/CWE/CWE-789/UnboundedAllocation.ql

```diff
@@ -0,0 +1,28 @@
+/**
+ * @name Unbounded allocation
+ * @kind path-problem
+ * @problem.severity warning
+ * @precision medium
+ * @id java/unbounded-allocation
+ * @tags security
+ *       external/cwe/cwe-789
+ */
+
+import java
+import UnboundedAllocationCommon
+import semmle.code.java.dataflow.FlowSources
+import DataFlow::PathGraph
+
+class UnboundedAllocationConfig extends TaintTracking::Configuration {
+  UnboundedAllocationConfig() { this = "UnboundedAllocationConfig" }
+
+  override predicate isSource(DataFlow::Node src) { src instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof AllocationSink }
+
+  override predicate isSanitizer(DataFlow::Node node) { hasUpperBound(node.asExpr()) }
+}
+
+from DataFlow::PathNode source, DataFlow::PathNode sink, UnboundedAllocationConfig config
+where config.hasFlowPath(source, sink)
+select sink, source, sink, "Unbounded memory allocation"
```

Comment:
I think we recommend that select statements include placeholders: https://github.com/github/codeql/blob/main/docs/query-metadata-style-guide.md#alert-messages